pyiceberg/transforms.py

```diff
@@ -517,9 +517,6 @@ def day_func(v: Any) -> int:
     def can_transform(self, source: IcebergType) -> bool:
         return isinstance(source, (DateType, TimestampType, TimestamptzType))
 
-    def result_type(self, source: IcebergType) -> IcebergType:
-        return DateType()
-
```

Comment:
nit: can we be explicit here and do 
```
    def result_type(self, source: IcebergType) -> IntegerType:
        return IntegerType()
```

Reply:
I'm matching the behavior of the other transforms that extend `TimeTransform`, which all just implicitly use `TimeTransform.result_type` instead of overriding it. Should we change this for all of them?

Reply:
gotcha, thanks for the context. This is fine!